### PR TITLE
US-33 Created link controller and jwt implementation

### PR DESCRIPTION
**Pull request by ilovepilav**

### CODE REVIEW

File: src/URLShortener.API/Controllers/LinksController.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    [UserType(UserType.User)]
+    public async Task<ICollection<Link>> Get()
+    {
+        var userClaim = Request.HttpContext.User.Claims.First(c => c.Type == JwtRegisteredClaimNames.Sid).Value;
+        Guid.TryParse(userClaim, out var userId);
+
+        return await _linkService.GetAllByUserIdAsync(userId);
+    }
+
+    [HttpGet("{linkId:long}")]
```
Comment (ilovepilav) [inline src/URLShortener.API/Controllers/LinksController.cs:32]: this needs to be fixed

File: src/URLShortener.API/Controllers/LinksController.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    [UserType(UserType.Super)]
+    public async Task<ICollection<Link>> Get(long linkId)
+    {
+        var userClaim = Request.HttpContext.User.Claims.First(c => c.Type == JwtRegisteredClaimNames.Sid).Value;
+        Guid.TryParse(userClaim, out var userId);
+
+        return await _linkService.GetAllByUserIdAsync(userId);
+    }
+
+    [HttpGet("{shortAddress}")]
```
Comment (ilovepilav) [inline src/URLShortener.API/Controllers/LinksController.cs:46]: template

File: src/URLShortener.API/Controllers/LinksController.cs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        });
+    }
+
+    [HttpPost]
+    [UserType(UserType.User)]
+    public async Task<Link> Create([FromBody] CreateLinkModel model)
+    {
+        var userClaim = Request.HttpContext.User.Claims.First(c => c.Type == JwtRegisteredClaimNames.Sid).Value;
+        Guid.TryParse(userClaim, out var userId);
+
```
Comment (ilovepilav) [inline src/URLShortener.API/Controllers/LinksController.cs:63]: userId

File: src/URLShortener.API/Program.cs [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+app.UseHttpsRedirection();
+
+app.UseAuthentication();
+
+app.UseAuthorization();
+
+app.MapControllers();
+
+using (var scope = app.Services.CreateScope())
```
Comment (ilovepilav) [inline src/URLShortener.API/Program.cs]: delete this

**Review by devSakaoglu — CHANGES_REQUESTED** [review]
Changes needed

**Review by devSakaoglu — APPROVED** [review]

**Pull request merged**